androidApp/src/main/java/com/mbta/tid/mbta_app/android/stopDetails/StopDetailsFilteredDeparturesView.kt

```diff
@@ -96,21 +100,147 @@ fun StopDetailsFilteredDeparturesView(
+        analytics
+    )
+}
+
+@Composable
+fun StopDetailsFilteredDeparturesView(
+    stopId: String,
+    stopFilter: StopDetailsFilter,
+    tripFilter: TripDetailsFilter?,
+    routeCardData: RouteCardData,
+    routeStopData: RouteCardData.RouteStopData,
+    leaf: RouteCardData.Leaf,
+    tileData: List<TileData>,
+    noPredictionsStatus: NoTripsFormat?,
+    allAlerts: AlertsStreamDataResponse?,
+    elevatorAlerts: List<Alert>,
+    global: GlobalResponse?,
+    now: Instant,
+    viewModel: StopDetailsViewModel,
+    errorBannerViewModel: ErrorBannerViewModel,
+    updateStopFilter: (StopDetailsFilter?) -> Unit,
+    updateTripFilter: (TripDetailsFilter?) -> Unit,
+    tileScrollState: ScrollState,
+    pinnedRoutes: Set<String>,
+    togglePinnedRoute: (String) -> Unit,
+    onClose: () -> Unit,
+    openModal: (ModalRoutes) -> Unit,
+    openSheetRoute: (SheetRoutes) -> Unit,
+    analytics: Analytics = koinInject()
+) {
+    Layout(
+        stopId,
+        stopFilter,
+        tripFilter,
+        groupByDirection = true,
+        routeCardData.lineOrRoute,
+        routeStopData.stop,
+        availableDirections = routeStopData.data.map { it.directionId }.distinct().sorted(),
+        routeStopData.directions,
+        tileData,
+        noPredictionsStatus,
+        selectedTripIsCancelled =
+            tripFilter?.let {
+                leaf.upcomingTrips.any { it.trip.id == tripFilter.tripId && it.isCancelled }
+            }
+                ?: false,
+        allAlerts,
+        leaf.alertsHere,
+        leaf.alertsDownstream,
+        elevatorAlerts,
+        global,
+        now,
+        viewModel,
+        errorBannerViewModel,
+        updateStopFilter,
+        updateTripFilter,
+        tileScrollState,
+        pinnedRoutes,
+        togglePinnedRoute,
+        onClose,
+        openModal,
+        openSheetRoute,
+        analytics
+    )
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+private fun Layout(
+    stopId: String,
+    stopFilter: StopDetailsFilter,
+    tripFilter: TripDetailsFilter?,
+    groupByDirection: Boolean,
+    lineOrRoute: RouteCardData.LineOrRoute,
+    stop: Stop,
+    availableDirections: List<Int>,
+    directions: List<Direction>,
+    tileData: List<TileData>,
+    noPredictionsStatus: NoTripsFormat?,
+    selectedTripIsCancelled: Boolean,
+    allAlerts: AlertsStreamDataResponse?,
+    alertsHere: List<Alert>,
+    downstreamAlerts: List<Alert>,
+    elevatorAlerts: List<Alert>,
+    global: GlobalResponse?,
+    now: Instant,
+    viewModel: StopDetailsViewModel,
+    errorBannerViewModel: ErrorBannerViewModel,
+    updateStopFilter: (StopDetailsFilter?) -> Unit,
+    updateTripFilter: (TripDetailsFilter?) -> Unit,
+    tileScrollState: ScrollState,
+    pinnedRoutes: Set<String>,
+    togglePinnedRoute: (String) -> Unit,
+    onClose: () -> Unit,
+    openModal: (ModalRoutes) -> Unit,
+    openSheetRoute: (SheetRoutes) -> Unit,
+    analytics: Analytics = koinInject()
```

Comment:
This is far, far too many parameters for one method, but it’s not that many more parameters than this component already had, and I can’t think of obvious ways to group a bunch of them together at once in a way that’s actually cohesive and not just cosmetic.

Reply:
I just wasn’t creative enough; most of these parameters are the same in either variant, so putting the ones that actually differ into a `sealed interface` lets me keep the parameter count the same and get rid of the inner helper function completely.